Enhance OA vignette and update descriptions (#750)
Olink-Proteomics/OlinkRPackage: OlinkAnalyze/vignettes/OlinkAnalyze-Vignette.Rmd

Reviewer: When will this vignette be published in relation to the publication of the OlinkAnalyzeVignettes package? If at/around the same time, is it worth removing that the package 'will be published soon'?

Author: I have a card about it. Will be fixed.

## Introduction to Olink NPX data format

The package contains two test data files named `npx_data1` and `npx_data2`.
These are synthetic datasets that resemble Olink® data accompanied by clinical
variables. Olink® data that is delivered in long format or imported with the
function `read_npx()` contain the following columns:

-   `SampleID` *\<chr\>*: Sample names or IDs.
-   `OlinkID` *\<chr\>*: Unique ID for each assay assigned by Olink. In case

>     the assay is included in more than one panels it will have a different

Reviewer: ```suggestion
    the assay is included in more than one panel it will have a different
```

-   `Panel` *\<chr\>*: Olink Panel that samples ran on. Read more about Olink
    Panels here: <https://olink.com/products/compare>.
-   `Panel_Version` *\<chr\>*: Version of the panel. A new panel version might
    include some different or improved assays.
-   `PlateID` *\<chr\>*: Name of the plate.
-   `QC_Warning` or `SampleQC` *\<chr\>*: Indication whether the sample passed
    Olink QC. More information about Olink quality control metrics can be
    found in our [FAQ](https://olink.com/knowledge/faq?query=quality%20control)
    by searching for the term "Quality control".
-   `LOD` *\<dbl\>*: Limit of detection (LOD) is the minimum level of an
    individual protein that can be measured. LOD is defined as 3 times the
    standard deviation over background.
-   `NPX` *\<dbl\>*: Normalized Protein eXpression (NPX), is Olink®’s unit of
    protein expression level in a log~2~ scale. The majority of the functions of
    this package use NPX values for calculations. Read more about NPX in the
    Olink [FAQ](https://olink.com/knowledge/faq?query=what%20is%20npx) (search
    term "What is NPX?") or in Olink's Data normalization and standardization

>     [white paper](https://7074596.fs1.hubspotusercontent-na1.net/hubfs/7074596/05-white%20paper%20for%20website/1096-olink-data-normalization-white-paper.pdf).

Reviewer: Confirmed that all hyperlinks in this section are live and lead to the correct webpages.

**Note:** There are 5 additional variables in the sample datasets `npx_data1`
and `npx_data2` that include clinical or other information, namely: Subject
*\<chr\>*, Treatment *\<chr\>*, Site *\<chr\>*, Time *\<chr\>*, Project
*\<chr\>*.

The columns found in an Olink data set may vary based on the version and
product.

# Preprocessing

## Read NPX data (`read_NPX()` or `read_npx()`)

> The `read_npx()` function imports an NPX file into a tidy format to work with in

Reviewer: ```suggestion
The `read_npx()` (or `read_NPX)`) function imports an NPX file into a tidy format to work with in
```

Reviewer: Optional suggestion, can leave as is if preferred.

R. This function supports Olink® NPX files generated by Olink® data software in
CSV, Excel, and Parquet formats. No prior alterations to the NPX output file
should be made for this function to work as expected.

### Function arguments

-   `filename`: Path to the NPX output file.
-   `out_df`: Output data frame format. Either "tibble" or "ArrowObject".
    Default: "tibble".
-   `long_format`: Logical. Whether the input file is in long or wide format.

>     Default: "TRUE".

Reviewer: Should the long_format parameter documentation have 'Default: "NULL"'?

See line 35: https://github.com/Olink-Proteomics/OlinkRPackage/blob/develop/OlinkAnalyze/R/read_npx.R

-   `olink_platform`: Olink product used to generate the input file. One of
    "Target 48", "Flex", "Target 96", "Explore 3072", "Explore HT", "Focus", or

>     "Reveal". Defaults to NULL for auto-detection.

Reviewer: _Optional suggestion_ for lines 191 and 193:
```suggestion
    "Reveal". Defaults to "NULL" for auto-detection.
```

-   `data_type`: Quantification method of the input data. One of "Ct", "NPX", or
    "Quantified". Defaults to NULL for auto-detection.
-   `.ignore_files`: Character vector of files included in the zip-compressed
    Olink software output files that should be ignored. Used only for
    zip-compressed input files.
-   `quiet`: Logical. Whether to suppress messages about the import process.
    Default: "TRUE".
-   `legacy`: Logical. Whether to use the legacy import method. Default:
    "FALSE". The legacy method is less efficient and will be deprecated in
    future versions.

```{r, message = FALSE, eval = FALSE}
data <- OlinkAnalyze::read_npx(
  filename = "~/NPX_file_location.xlsx"
)

# OR
data <- OlinkAnalyze::read_NPX(
  filename = "~/NPX_file_location.xlsx"
)
```

### Function output

A `tibble` or `ArrowObject` in long format containing:

-   `SampleID`: Sample names or IDs.
-   `OlinkID`: Unique ID for each assay assigned by Olink. In case the assay is

>     included in more than one panels it will have a different `OlinkID` in each

Reviewer: ```suggestion
    included in more than one panel it will have a different `OlinkID` in each
```

    one.
-   `UniProt`: UniProt ID.
-   `Assay`: Common gene name for the assay.

> -   `MissingFreq`: Missing frequency for the `OlinkID`, i.e. frequency of

Reviewer: Is this true for NGS-based products since we don't report LOD by default?

-   `Panel_Version`: Version of the panel. A new panel version might include
    some different or improved assays.
-   `PlateID`: Name of the plate.
-   `QC_Warning` or `SampleQC`: Indication whether the sample passed Olink QC.
    More information about Olink quality control metrics can be found in our
    [FAQ](https://olink.com/knowledge/faq?query=quality%20control) (search term
    "Quality control").
-   `LOD`: Limit of detection (LOD) is the minimum level of an individual
    protein that can be measured. LOD is defined as 3 times the standard
    deviation over background.
-   `NPX`: Normalized Protein eXpression (NPX), is Olink’s unit of protein
    expression level in a log~2~ scale. The majority of the functions of this
    package use NPX values for calculations. Read more about NPX in the Olink
    [FAQ](https://olink.com/knowledge/faq?query=what%20is%20npx) (search term
    "What is NPX?") or in Olink's Data normalization and standardization
    [white paper](https://7074596.fs1.hubspotusercontent-na1.net/hubfs/7074596/05-white%20paper%20for%20website/1096-olink-data-normalization-white-paper.pdf).

## Read multiple NPX data files (`read_NPX()` or `read_npx()`)

> In order to import multiple NPX data files at once, the `read_npx() function can

Reviewer: ```suggestion
In order to import multiple NPX data files at once, the `read_npx() function can`
```

  full.names = TRUE
) |>
  lapply(
    OlinkAnalyze::read_npx
  )  |>
  dplyr::bind_rows()
```

## Check NPX data quality (`check_npx()`)

The `check_npx()` function performs various quality and format checks on NPX
data imported with `read_npx()`. It is recommended to run this function after
reading in NPX data and before downstream analysis. The result can be passed as
the `check_log` argument to `clean_npx()` and all downstream Olink Analyze
functions, allowing each function to skip its own internal check and improve
performance.

`check_npx()` also allows for alternative column names to be used in the
downstream analysis, both standard and non-standard. If the user wishes to
analyze data with non-standard column names, or to resolve column name
ambiguities, they can pass a named character vector to the `preferred_names`
argument. For example, one might wish to use the column "PCNormalizedNPX"
instead of "NPX" for downstream analysis, or they might have log~2~ transformed
the values reported in "Quantified_value" and wish to use that column to analyze 

> their data. **Note** that for the time being, not all functions utilize the full

Reviewer: _Optional suggestion:_ move this note as a separate paragraph.
```suggestion
their data. **Note:** for the time being, not all functions utilize the full
```

capacity of the `preferred_names` argument, but we aim to expand this in the
upcoming releases.

### Function arguments

-   `df`: NPX data frame in long format, as returned by `read_npx()`.
-   `preferred_names`: Optional named character vector to resolve column name
    ambiguities or to map custom column names to internally expected ones.

```{r, message = FALSE, eval = FALSE}
# Check NPX data quality and format
check_log <- OlinkAnalyze::check_npx(
  df = data
)
```

### Function output

A named list with the following elements:

-   `col_names` *\<list\>*: Column names from the input data frame to be used in
    downstream analyses.
-   `oid_invalid` *\<chr\>*: OlinkID values that do not follow the expected

>     format (OID#####).

Reviewer: I know we are expanding the allowed format to follow OID#####_OID##### for cross-product bridged datasets. 

Not sure if worth mentioning this here? I'm okay with leaving the text as-is if preferred.

Author: Good catch. Fixed.

-   `remove_qc_warning`: Logical. Remove samples with QC status "FAIL". Default:
    "TRUE".
-   `remove_assay_warning`: Logical. Remove assays flagged with assay warnings.
    Default: "TRUE".
-   `control_sample_ids`: Character vector of identifiers of samples that should
    be removed. Default: "NULL".
-   `convert_df_cols`: Logical. Convert columns to their expected data types.
    Default: "TRUE".
-   `convert_nonunique_uniprot`: Logical. Resolve non-unique "OlinkID"-"UniProt"
    mappings. Default: "TRUE".
-   `verbose`: Logical. Print progress messages. Default: "FALSE".

```{r, message = FALSE, eval = FALSE}
# Clean the NPX data using the check_npx output
data_clean <- OlinkAnalyze::clean_npx(
  df = data,
  check_log = check_log
)
```

### Function output

A "tibble" (or "ArrowObject") in long format containing the cleaned NPX data,
with invalid assays, control samples, QC-failing samples, and problematic
entries removed according to the chosen arguments. Output matches that of the
function `read_npx()`.

**Note:** We recommend running `check_npx()` once again after cleaning the data
to confirm that all issues have been resolved and that the data is ready for

> downstream analysis. An schematic illustration of the described workflow is

Reviewer: ```suggestion
downstream analysis. A schematic illustration of the described workflow is
```

> provided ["Olink Analyze v5.0 example workflow"](#usage).

Reviewer: ```suggestion
provided ([Olink Analyze v5.0 example workflow](#usage)).
```

> ```{r message=FALSE, eval=FALSE}

Reviewer: ```suggestion
```{r message = FALSE, eval = FALSE}
```

-   **Subset normalization**: A subset of samples is used to normalize two data
    sets, one of which is used as a `reference_project`. Adjustment is made
    using the differences of medians between the sample subsets from the two
    data sets. Subset normalization is useful if no bridge samples were included
    and one can assume that the distribution of the two datasets is very similar.
-   **Reference median normalization**: Works only on one data set. This is
    effectively subset normalization, but using difference of medians to
    pre-recorded median values. `df1`, `overlapping_samples_df1` and
    `reference_medians` need to be specified.
-   **Cross-product bridging**: Similar to bridging normalization but bridging
    across products, for example bridging "Olink Explore 3072" data to "Olink
    Explore HT" data. Overlapping samples are run on both products and used to
    determine which assays are bridgeable and what method should be used to
    bridge each assay. For more information on the between-product bridging
    methodology, consult the **Bridging across NGS-based Olink^®^ products**
    **Tutorial** in the R package `OlinkAnalyzeVignettes`.

> ## Integrating Explore NPX LOD (`olink_lod()`)

Reviewer: May be worth addressing in a future release if time does not permit - should we use NGS instead of Explore?

The LOD vignette referenced on line 476 also uses Explore in the title. Can this be updated to NGS at some point as well?

Author: Good catch. Fixed it in this vignette, and made a card for it to be fixed in the LOD vignette.

Crossed/interaction analysis, i.e. A\*B formula notation, is inferred from the
variable argument in the following cases:

-   c('A', 'B')
-   c('A:B')
-   c('A:B', 'B') or c('A:B', 'A')

For covariates, crossed analyses need to be specified explicitly, i.e. two main
effects will not be expanded with a `c('A', 'B')` notation. Main effects present
in the variable take precedence.

Adjusted p-values are calculated using the Benjamini & Hochberg (1995) method
("fdr"). The threshold is determined by logic evaluation of `Adjusted_pval` \<
0.05. Covariates are not included in the p-value adjustment.

### Function arguments

-   `df`: NPX data frame in long format should minimally contain protein name
    (`Assay`), `OlinkID`, `UniProt`, `Panel` and an outcome factor with at least
    3 levels.
-   `variable`: Single character value or character array. In case of single

>     character then that should represent a column in the `df`. Otherwise, if

Reviewer: ```suggestion
    character, then that should represent a column in the `df`. Otherwise, if
```

-   c('A', 'B')
-   c('A:B')
-   c('A:B', 'B') or c('A:B', 'A')

For covariates, crossed analyses need to be specified explicitly, i.e. two main
effects will not be expanded with a c('A', 'B') notation. Main effects present
in the variable take precedence.

Adjusted p-values are calculated using the Benjamini & Hochberg (1995) method
("fdr"). The threshold is determined by logic evaluation of Adjusted_pval \<
0.05. Covariates are not included in the p-value adjustment.

### Function arguments

-   `df`: NPX data frame in long format should minimally contain protein name
    (`Assay`), `OlinkID`, `UniProt`, `Panel` and 1-2 variables with at least 2
    levels and subject identifiers (`SubjectID`).
-   `variable`: Single character value or character array. In case of single
    character then that should represent a column in the `df`. Otherwise, if
    length \> 1, the included variable names will be used in crossed analyses.
    It can also accept the notations ':' or '\*'.
-   `outcome`: Name of the column from `df` that contains the dependent
    variable. Default: "NPX".
-   `random`: Single character value or character array with random effects.

> -   `covariates`: Single character value or character array. Default: `NULL`.

Reviewer: _Optional suggestion:_ should the default arguments have double quotations to match the format for 'outcome' and other previous functions in the vignette (e.g. read_npx() and check_npx())?

See the following parameters:
- covariates
- return.covariates
- verbose

Author: No. These quotations "`" are different from the single/double quotations. The first one usually symbolizes programmatically relevant objects `NULL`, `TRUE`, `filename` (argument of a function, e.g. read_npx(filename = "my_file")), while the latter marks column names, or simple strings of characters. Also, in R single a double quotation are interchangeable.

I fixed this to be consistent, as there were some "TRUE", "FALSE" and "NULL" across the vignette.

deviation difference in the numerical variable, e.g. mean NPX at mean(numerical
variable) versus mean NPX at mean(numerical variable) + 1\*SD(numerical
variable). The output tibble is arranged by ascending adjusted p-values.

### Function arguments

-   `df`: NPX data frame in long format should minimally contain protein name
    (`Assay`), `OlinkID`, `UniProt`, `Panel` and 1-2 variables with at least 2
    levels and subject identifiers (`SubjectID`).
-   `variable`: Single character value or character array. In case of single
    character then that should represent a column in the `df`. Otherwise, if
    length \> 1, the included variable names will be used in crossed analyses.
    It can also accept the notations ':' or '\*'.
-   `olinkid_list`: Character vector of `OlinkID`'s on which to perform the
    post-hoc analysis. If not specified, all assays in `df` are used.
-   `effect`: Character vector. Term on which to perform the post-hoc analysis.
    Must be subset of or identical to the variable.
-   `outcome`: Name of the column from `df` that contains the dependent
    variable. Default: "NPX".
-   `random`: Single character value or character array with random effects.

> -   `covariates`: Single character value or character array. Default: `NULL`.

Reviewer: _Optional suggestion:_ should the default arguments have double quotations to match other previous functions in the vignette (e.g. read_npx() and check_npx())?

See the following parameters:

- covariates
- verbose

Author: See above.

    calculation. The ":" between variables indicates interaction between
    variables.
-   `contrast` *\<chr\>*: Variables (in term) that are compared.
-   `estimate` *\<dbl\>*: Difference in mean NPX between variables (from
    contrast).
-   `conf.low` *\<dbl\>*: Low bound of the confidence interval for the mean.
-   `conf.high` *\<dbl\>*: High bound of the confidence interval for the mean.
-   `Adjusted_pval` *\<dbl\>*: Adjusted p-value for the test (Benjamini &
    Hochberg).
-   `Threshold` *\<chr\>*: Text indication if assay is significant (adjusted
    p-value \< 0.05).

## Additional Statistical Tests

Many other statistical functions can be found within Olink Analyze, including:

-   `olink_one_non_parametric()` Function which performs a Kruskal-Wallis Test
    or Friedman Test per protein.
-   `olink_one_non_parametric_posthoc()` Function which performs post-hoc test
    for one way non-parametric test.
-   `olink_ordinalRegression()` Function which performs an ordinal regression
    per protein.
-   `olink_ordinalRegression_posthoc()` Function which performs an ordinal
    regression post-hoc test per protein.

> To learn more about these function, consult their help documentation using the

Reviewer: ```suggestion
To learn more about these functions, consult their help documentation using the
```

`help()` function.

## Pathway Enrichment (`olink_pathway_enrichment()`)

The `olink_pathway_enrichment()` function can be used to perform Gene Set

> Enrichment Analysis (GSEA) or Over-representation Analysis (ORA) using MSigDB,

Reviewer: ```suggestion
Enrichment Analysis (GSEA) or Over-Representation Analysis (ORA) using MSigDB,
```

This function performs enrichment using the `gsea()` or `enrich()` functions
from the `clusterProfiler` R package from BioConductor. The function uses the
estimate from a previous statistical analysis for one contrast for all proteins.
MSigDB is subset if ontology is KEGG, GO, or Reactome. `test_results` must
contain estimates for all assays. Post-hoc results can be used but should be
filtered for one contrast to improve interpretability.

Alternative statistical results can be used as input as long as they include the
columns `OlinkID`, `Assay`, and `estimate`. A column named `Adjusted_pval` is
also needed for ORA. Any statistical results that contains one estimate per
protein will work as long as the estimates are comparable to each other.

### Function Arguments

-   `df`: NPX data frame in long format with columns `Assay`, `OlinkID`,
    `UniProt`, `SampleID`, `QC_Warning` or `SampleQC`, and `NPX`.
-   `test_results`: a data frame of statistical test results including
    `Adjusted_pval` and estimate columns.
-   `method`: String of method name. Must be either "GSEA" (default) or "ORA".
-   `ontology`: String of database to query. Must be either "MSigDb", "KEGG",

>     "GO", and "Reactome".

Reviewer: ```suggestion
    "GO", or "Reactome".
```

    normalized enrichment score represents a false positive finding.
-   `rank` *\<dbl\>*: the position in the ranked list where the maximum
    enrichment score occurred.
-   `leading_edge` *\<chr\>*: contains tags, list, and signal. Tags gives an
    indication of the percentage of genes contributing to the enrichment score.
    List gives an indication of where in the list the enrichment score is
    obtained. Signal represents the enrichment signal strength and combines the
    tag and list.
-   `core_enrichment` *\<chr\>*: list of input proteins (Gene Symbols) annotated
    in a term delimited by "/".

# Exploratory analysis

## Principal components analysis (PCA) plot (`olink_pca_plot()`)

Generates PCA projection of all samples from NPX data along two principal
components (default 'PC1' vs 'PC2') colored by the variable specified by
`color_g` (default 'QC_Warning') and including the percentage of explained
variance. By default, the values are scaled and centered in the PCA and proteins
with missing NPX values removed from the corresponding assays. Unique sample
names are required. Imputation by median value is performed for assays with

> missingness \<10% and for multi-plate projects, and for missingness \<5% for

Reviewer: ```suggestion
missingness \<10% for multi-plate projects, and for missingness \<5% for
```

single plate projects.

More information about `olink_pca()` can be found in the **Outlier Exclusion**
**Vignette** in the R package `OlinkAnalyzeVignettes`.

## Uniform Manifold Approximation and Projection for Dimension Reduction (UMAP) (`olink_umap_plot()`)

Computes a manifold approximation and projection and plots the two specified
components. Unique sample names are required and imputation by the median is
done for assays with missingness \<10% for multi-plate projects and \<5% for
single plate projects.

The arguments `outlierDefX` and `outlierDefY` can be used to identify outliers
in the UMAP results. Sample outliers will be labelled.

> NOTE: UMAP is a non-linear data transformation that might not accurately

Reviewer: ```suggestion
**Note:** UMAP is a non-linear data transformation that might not accurately
```

Reviewer: This format matches similar notes in the vignette.

preserve the properties of the data. Distances in the UMAP plane should
therefore be interpreted with caution.

### Function arguments (selection)

-   `df`: NPX data frame in long format should minimally contain `SampleID`,
    `NPX` and column that will be used for grouping/coloring.
-   `color_g`: Character value indicating the column name that should be used as
    fill color. Default `QC_Warning`.
-   `x_val`: Integer indicating which principal component to plot along the
    x-axis. Default 1.
-   `y_val`: Integer indicating which principal component to plot along the
    y-axis. Default 2.
-   `config`: Object of class `umap.config`, specifying the parameters for the
    UMAP algorithm.
-   `label_samples`: Logical. If `TRUE`, points are replaced with `SampleID`.

>     Default `FALSE`.

Reviewer: _Optional suggestion:_ should the default arguments have double quotations to match other previous functions in the vignette (e.g. read_npx() and check_npx())? For consistency, we should also add a colon between 'Default' and the default argument

See the following parameters:

- color_g
- label_smples
- byPanel
- OutlierLines
- verbose
- quiet

Author: See above.

A list of objects of class `ggplot` is silently returned. Plots are also printed
unless option `quiet = TRUE` is set.

# Visualization

## Boxplots for outcomes (`olink_boxplot()`)

The `olink_boxplot()` function is used to generate boxplots of NPX values
stratified on a variable for a given list of proteins. In order to annotate the
plot with ANOVA posthoc analysis results (i.e. include statistical asterisks in
the plot), control samples and control assays should be removed from the data.

### Function arguments

-   `df`: NPX data frame in long format should minimally contain protein name
    (`Assay`), `OlinkID`, `UniProt` and a grouping variable.
-   `variable`: Single character value indicating the column name to use as a
    grouping variable in the x axis.
-   `olinkid_list`: Character vector of OlinkID's that should be used for the
    boxplot. If not specified, all assays in `df` are used.
-   `posthoc_results`: Data frame from ANOVA posthoc analysis. This data frame
    need to be generated using the `olink_anova_posthoc()` function.
-   `ttest_results`: Data frame from t-test analysis. This data frame need to be
    generated using the `olink_ttest()` function.

> -   `verbose`: Logical. Flag indicating if plots shall be printed additionally

Reviewer: _Optional suggestion:_ should the default arguments have double quotations to match other previous functions in the vignette (e.g. read_npx() and check_npx())?

See the following parameters:

covariates
- verbose

Author: See above.

    to assigned to a list variable. Default: `FALSE`.
-   `number_of_proteins_per_plot`: Number of boxplots to include in the facets
    plot. Default 6.
-   `check_log`: Named list returned by `check_npx()`. If `NULL`, `check_npx()`
    is run internally.

```{r, message = FALSE, eval = FALSE}
plot <- data_clean |>

>   # removing missing values which exist for Site

Reviewer: ```suggestion
  # removing missing values that exist for Site
```

  dplyr::filter(
    !is.na(.data[["Site"]])
  ) |>
  OlinkAnalyze::olink_boxplot(
    variable = "Site",
    olinkid_list = c("OID00488", "OID01276"),
    number_of_proteins_per_plot = 2L,
    check_log = check_log_clean
  )

plot[[1L]]
```

```{r, message = FALSE, echo = FALSE}
knitr::include_graphics(
  path = normalizePath(
    path = "../man/figures/olink_boxplot.png"
  ),
  error = FALSE
)
```

```{r, message = FALSE, eval = FALSE}
anova_posthoc_results <- OlinkAnalyze::olink_anova_posthoc(
  df = data_clean,
  olinkid_list = c("OID00488", "OID01276"),
  variable = "Site",
  effect = "Site",
  check_log = check_log_clean
)

plot2 <- data_clean |>

>   tidyr::drop_na() |> # removing missing values which exists for Site

Reviewer: ```suggestion
  tidyr::drop_na() |> # removing missing values that exist for Site
```

More information about `olink_dist_plot()` can be found in the **Outlier**
**Exclusion Vignette** in the R package `OlinkAnalyzeVignettes`.

## Point-range plot for LMER (`olink_lmer_plot()`)

The function `olink_lmer_plot()` generates a point-range plot for a given list
of proteins based on linear mixed effect model. The points illustrate the mean
NPX level for each group and the error bars illustrate 95% confidence intervals.
Facets are labeled by the protein name and corresponding OlinkID for the
protein.

### Function arguments

-   `df`: NPX data frame in long format should minimally contain protein name
    (`Assay`), `OlinkID`, `UniProt`, `Panel` and 1-2 variables with at least 2
    levels and subject ID (`SubjectID`).
-   `variable`: Single character value or character array. In case of single
    character then that should represent a column in the `df.` Otherwise, if
    length \> 1, the included variable names will be used in crossed analyses.
    It can also accept the notations ':' or '\*'.
-   `outcome`: Name of the column from `df` that contains the dependent
    variable. Default: "NPX".
-   `random`: Single character value or character array with random effects.

> -   `covariates`: Single character value or character array. Default: `NULL.`

Reviewer: _Optional suggestion:_ should the default arguments have double quotations to match other previous functions in the vignette (e.g. read_npx() and check_npx())?

See the following parameters:
- covariates
- verbose

Author: See above.

    specified, displays top terms.
-   `number_of_terms`: number of terms to display, default is 20.

### Function output

A bar graph as a `ggplot` object.

## Scatterplot for QC (`olink_qc_plot()`)

The `olink_qc_plot()` function generates a plot faceted by `Panel`, plotting IQR
vs. median NPX for all samples. This is a good first check to find out if any
samples have a tendency to be classified as outliers. Horizontal dashed lines
indicate +/-3 standard deviations from the mean IQR. Vertical dashed lines
indicate +/-3 standard deviations from the mean sample median.

More information about `olink_qc_plot()` can be found in the **Outlier**
**Exclusion Vignette** in the R package `OlinkAnalyzeVignettes`.

## Heatmap (`olink_heatmap_plot()`)

The `olink_heatmap_plot()` function generates a heatmap for a specfied set of
samples and proteins. By default, the heatmap centers and scales NPX across
proteins and clusters samples and proteins using a dendrogram. Unique sample
names are required.

The grouping variable(s) are annotated and colored in the left side of the
heatmap.

### Function arguments

-   `df`: NPX data frame in long format which should minimally contain

>     `SampleID`, `NPX`, `OlinkID`, `Assay.` Optionally columns of choice for

Reviewer: ```suggestion
    `SampleID`, `NPX`, `OlinkID`, `Assay.` Optionally, columns of choice for
```

    annotations.
-   `variable_row_list`: Columns in `df` to be annotated for rows in the
    heatmap.
-   `variable_col_list`: Columns in `df` to be annotated for columns in the
    heatmap.

> -   `center_scale`: Logical. If data should be centered and scaled across 

Reviewer: _Optional suggestion:_ should the default arguments have double quotations to match other previous functions in the vignette (e.g. read_npx() and check_npx())?

See the following parameters:
- center_scale
- cluster_rows
- cluster_cols
- show_rownames
- show_colnames
- annotation_legend

Author: See above.

)
```

```{r, message = FALSE, echo = FALSE}
knitr::include_graphics(
  path = normalizePath(
    path = "../man/figures/olink_volcano_plot.png"
  ),
  error = FALSE
)
```

### Function output

An object of class `ggplot`.

## Theming function (`set_plot_theme()`)

> This function sets a coherent plot theme for plots by adding it to a `ggplot`

Reviewer: ```suggestion
This function sets a consistent plot theme for plots by adding it to a `ggplot`
```

  OlinkAnalyze::set_plot_theme()
```

```{r, message = FALSE, echo = FALSE}
knitr::include_graphics(
  path = normalizePath(
    path = "../man/figures/set_plot_theme_boxplot.png"
  ),
  error = FALSE
)
```

## Color theming (`olink_color_discrete()`, `olink_color_gradient()`, `olink_fill_discrete()`, `olink_fill_gradient()`)

> These functions sets a coherent coloring theme for the plots by adding it to a

Reviewer: ```suggestion
These functions set a consistent coloring theme for the plots by adding it to a
```

**products Tutorial** in the R package `OlinkAnalyzeVignettes`.

### Function arguments

-   `df`: NPX data frame generated from between-product bridging in long format.
    Should minimally contain `Assay`, `OlinkID`, `OlinkID_E3072`, and all data
    points corresponding to bridging samples from the reference project and the
    new project.
-   `median_counts_threshold`: Integer indicating minimum median counts allowed
    for each platform. If either platform has median counts below 150 for an
    assay, the assay fails the counts criteria when evaluating bridgeability.
    Default: 150.
-   `min_counts`: Integer indicating minimum counts allowed for a data point. If
    any data point in the bridge normalized dataframe contains fewer than the
    defined minimum count cutoff, it is excluded from the bridgeability
    assessment and corresponding figures. Default: 10.
-   `bridge_sampleid`: Character vector containing overlapping SampleIDs between
    the two bridging projects. If this argument is not provided, the function
    will look for overlapping `SampleID` values between the two projects in the

>     bridged dataframe. Default: `NULL`.

Reviewer: _Optional suggestion:_ should the default arguments have double quotations to match other previous functions in the vignette (e.g. read_npx() and check_npx())?

See the following parameters:
- bridge_sampleid

Author: See above.

> ```{r message=FALSE, echo=FALSE, out.width="600px"}

Reviewer: ```suggestion
```{r, message = FALSE, echo = FALSE, out.width = "600px"}
```
